src/OrleansRuntime/MultiClusterNetwork/MultiClusterOracle.cs

```diff
@@ -0,0 +1,795 @@
+
+            PublishChanges();
+
+            // wait for the gossip channel tasks and aggregate exceptions
+            var currentChannelTasks = this.channelTasks.Values.ToList();
+            await Task.WhenAll(currentChannelTasks.Select(ct => ct.Task));
+
+            var exceptions = currentChannelTasks
+                .Where(ct => ct.LastException != null)
+                .Select(ct => ct.LastException)
+                .ToList();
+
+            logger.Info("Completed MultiClusterConfiguration Injection, {0} exceptions", exceptions.Count);
+
+            if (exceptions.Count > 0)
+                throw new AggregateException(exceptions);
+        }
+
+        public void SiloStatusChangeNotification(SiloAddress updatedSilo, SiloStatus status)
+        {
+            // any status change can cause changes in gateway list
+            PublishChanges();
+        }
+
+        public async Task Start(ISiloStatusOracle oracle)
+        {
+            logger.Info(ErrorCode.MultiClusterNetwork_Starting, "MultiClusterOracle starting on {0}, Severity={1} ", Silo, logger.SeverityLevel);
+            try
+            {
+                if (string.IsNullOrEmpty(clusterId))
+                    throw new OrleansException("Internal Error: missing cluster id");
+
+                this.siloStatusOracle = oracle;
+
+                // startup: pull all the info from the tables, then inject default multi cluster if none found
+                foreach (var ch in gossipChannels)
+                {
+                    SynchronizeWithChannel(ch);
+                }
+
+                await Task.WhenAll(this.channelTasks.Select(kvp => kvp.Value.Task));
+                if (GetMultiClusterConfiguration() == null && defaultMultiCluster != null)
+                {
+                    this.injectedConfig = new MultiClusterConfiguration(DateTime.UtcNow, defaultMultiCluster, "DefaultMultiCluster");
+                    logger.Info("No configuration found. Using default configuration {0} ", this.injectedConfig);
+                }
+
+                this.siloStatusOracle.SubscribeToSiloStatusEvents(this);
+
+                PublishChanges();
+
+                StartTimer(); // for periodic full bulk gossip
+
+                logger.Info(ErrorCode.MultiClusterNetwork_Starting, "MultiClusterOracle started on {0} ", Silo);
+            }
+            catch (Exception exc)
+            {
+                logger.Error(ErrorCode.MultiClusterNetwork_FailedToStart, "MultiClusterOracle failed to start {0}", exc);
+                throw;
+            }
+        }
+
+        private void StartTimer()
+        {
+            if (timer != null)
+                timer.Dispose();
+
+            timer = GrainTimer.FromTimerCallback(
+                this.OnGossipTimerTick,
+                null,
+                this.randomizedBackgroundGossipInterval,
+                this.randomizedBackgroundGossipInterval,
```

Comment:
It was at most 10% shorter (randomization subtracts between 0% and 10%), so I don't think it would be a big deal for load.

But I don't really have a problem with going the other way to be conservative, so  I will randomize by adding between 0% and 5%.